snapcraft/plugins/dump.py

```diff
@@ -35,7 +35,58 @@ class DumpPlugin(snapcraft.BasePlugin):
 
     def build(self):
         super().build()
-        if os.path.exists(self.installdir):
-            os.rmdir(self.installdir)
-        shutil.copytree(self.builddir, self.installdir,
-                        copy_function=snapcraft.common.link_or_copy)
+        _linktree(self.builddir, self.installdir, self.installdir)
+
+
+def _link_or_copy(source, destination, boundary):
```

Comment:
a nit from the clean code book. Define the functions in this order:

_link_tree
_link_or_copy
_create_similar_directory

That way you can read it from top to bottom, with less vertical jumps.